src/wp-includes/class-wp-tax-query.php

```diff
@@ -633,10 +633,15 @@ public function transform_query( &$query, $resulting_field ) {
 				$args['term_taxonomy_id'] = $terms;
 				break;
 			default:
-				$args['include'] = wp_parse_id_list( $terms );
+				$terms           = wp_parse_id_list( $terms );
+				$args['include'] = $terms;
```

Comment:
So I can't count the args below, because depending on query, if it a different field. Making the logic a little more complex. The reason, this is done, is because `wp_parse_id_list` does a `array_unique`. Meaning if you submit `array( 3, 3, 3,)` this would become `array( 3 )`. Meaning the count is different. I think we should leave this as it is.

Reply:
> So I can't count the args below, because depending on query, if it a different field.

I'm unsure what you mean, are you able to clarify?

Reply:
@spacedmonkey I agree with @peterwilsoncc's concern here. I think it's safer to not modify the value of `$terms` and keep the `wp_parse_id_list( $terms )` directly within `$args['include']`. I've also added a suggestion below for how you could still address the count bit below.

Reply:
I don't think either of you have understood the code here. Terms is an array, of ids, or slugs or names. This array is passed to WP_Term_Query and the count of the array should be the limit.

Reply:
1. Why is a problem to change the $term variable. This variable is not re-used apart from my change, so is safe to change. 
2. Terms is used to set term_taxonomy_id / name / slug args. We can't use includes, as this would only cover the default case and not the all other cases.